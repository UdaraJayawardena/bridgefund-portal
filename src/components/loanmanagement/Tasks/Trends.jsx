import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { Link } from 'react-router-dom';
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
import Table from "@material-ui/core/Table";
import TableRow from "@material-ui/core/TableRow";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
/*** CFO Imports */
import ArrowUpward from "@material-ui/icons/ArrowUpward";
import ArrowDownward from "@material-ui/icons/ArrowDownward";
import TableHead from '@material-ui/core/TableHead';
/**************/
// core components
import tasksStyle from "assets/jss/material-dashboard-react/components/tasksStyle.jsx";

class Trends extends React.Component {
  state = {
    checked: this.props.checkedIndexes
  };
  handleToggle = value => () => {
    const { checked } = this.state;
    const currentIndex = checked.indexOf(value);
    const newChecked = [...checked];

    if (currentIndex === -1) {
      newChecked.push(value);
    } else {
      newChecked.splice(currentIndex, 1);
    }

    this.setState({
      checked: newChecked
    });
  };
  render() {
    const { classes, trends, rtlActive } = this.props;
    const tableCellClasses = classnames(classes.tableCell, {
      [classes.tableCellRTL]: rtlActive
    })

    return (
      <Table className={classes.table}>
        <TableHead className={classes.tableRow}>
          <TableRow key={'tasks_header'} className={classes.tableRow}>
            <TableCell className={tableCellClasses}>
              {'Customer'}
            </TableCell>
            <TableCell className={tableCellClasses}>
              {'Percentage Behind'}
            </TableCell>
            <TableCell className={tableCellClasses}>
              {'Trend'}
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {trends.map((trend, key) => (
            <TableRow key={trend.customerId + '-' + key} className={classes.tableRow}>

              <TableCell className={tableCellClasses}>
                <Link to={`smeProfile/${trend.smeId}`} >
                  {trend.customerName}
                </Link>
              </TableCell>

              <TableCell className={tableCellClasses}>
                {trend.percentageBehind.toFixed(1)}%
              </TableCell>

              <TableCell className={classes.tableActions}>

                {trend.trend === 'up' ? (
                  <ArrowUpward
                    className={
                      classes.upArrowCardCategory + " " + classes.successText
                    }
                  />
                ) : (
                    <ArrowDownward
                      className={
                        classes.upArrowCardCategory + " " + classes.successText
                      }
                    />
                  )}

              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    );
  }
}

Trends.propTypes = {
  classes: PropTypes.object.isRequired,
  tasksIndexes: PropTypes.arrayOf(PropTypes.number),
  trends: PropTypes.array,
  checkedIndexes: PropTypes.array,
  rtlActive: PropTypes.bool
};

export default withStyles(tasksStyle)(Trends);
