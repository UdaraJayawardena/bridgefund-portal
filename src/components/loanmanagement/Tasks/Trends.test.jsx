import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer'
import Trends from 'components/loanmanagement/Tasks/Trends.jsx';

it('renders Trends component as User without crashing', () => {
  shallow(<Trends />);
});
describe('Trends --- Snapshot',()=>{
      const trendsData = [
          {
              id: 1,
              customerName: 'ABC',
              trend: 'up',
              percentageBehind: 1
          },
          {
              id: 2,
              customerName: 'ABC 123',
              trend: 'down',
              percentageBehind: 2
          },
          {
              id: 3,
              customerName: '456 xxx-01',
              trend: 'down',
              percentageBehind: 3
          }
      ];
      let trend_up = [], trend_down = [], trend_up_indexes = [], trend_down_indexes = [];

      trend_up = trendsData.filter(e => e.trend == 'up');
      trend_down = trendsData.filter(e => e.trend == 'down');

      Object.keys(trend_up).map((e, key) =>
        trend_up_indexes.push(key)
      );
      Object.keys(trend_down).map((e, key) =>
        trend_down_indexes.push(key)
      );
      test('+++capturing Snapshot of Trends Up', () => {
          const renderedValue =  renderer.create(<Trends trends={trendsData} tasksIndexes={trend_up_indexes}/>).toJSON()
          expect(renderedValue).toMatchSnapshot();
      });

      test('+++capturing Snapshot of Trends Down', () => {
        const renderedValue =  renderer.create(<Trends trends={trendsData} tasksIndexes={trend_down_indexes}/>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});