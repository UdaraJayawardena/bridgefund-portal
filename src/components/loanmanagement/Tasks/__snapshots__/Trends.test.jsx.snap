// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard --- Snapshot +++capturing Snapshot of Dashboard 1`] = `
<table
  className="MuiTable-root-18 Trends-table-9"
>
  <thead
    className="MuiTableHead-root-19 Trends-tableRow-10"
  >
    <tr
      className="MuiTableRow-root-20 MuiTableRow-head-23 Trends-tableRow-10"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Customer
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Percentage Behind
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Trend
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-37"
  >
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        1.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Dashboard --- Snapshot +++capturing Snapshot of Trends 1`] = `
<table
  className="MuiTable-root-18 Trends-table-9"
>
  <thead
    className="MuiTableHead-root-19 Trends-tableRow-10"
  >
    <tr
      className="MuiTableRow-root-20 MuiTableRow-head-23 Trends-tableRow-10"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Customer
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Percentage Behind
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Trend
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-37"
  >
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        1.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Dashboard --- Snapshot +++capturing Snapshot of Trends Down 1`] = `
<table
  className="MuiTable-root-18 Trends-table-9"
>
  <thead
    className="MuiTableHead-root-19 Trends-tableRow-10"
  >
    <tr
      className="MuiTableRow-root-20 MuiTableRow-head-23 Trends-tableRow-10"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Customer
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Percentage Behind
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Trend
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-37"
  >
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        1.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC 123
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        2.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Trends --- Snapshot +++capturing Snapshot of Trends 1`] = `
<table
  className="MuiTable-root-18 Trends-table-9"
>
  <thead
    className="MuiTableHead-root-19 Trends-tableRow-10"
  >
    <tr
      className="MuiTableRow-root-20 MuiTableRow-head-23 Trends-tableRow-10"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Customer
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Percentage Behind
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Trend
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-37"
  >
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        1.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Trends --- Snapshot +++capturing Snapshot of Trends Down 1`] = `
<table
  className="MuiTable-root-18 Trends-table-9"
>
  <thead
    className="MuiTableHead-root-19 Trends-tableRow-10"
  >
    <tr
      className="MuiTableRow-root-20 MuiTableRow-head-23 Trends-tableRow-10"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Customer
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Percentage Behind
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Trend
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-37"
  >
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        1.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </td>
    </tr>
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC 123
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        2.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Trends --- Snapshot +++capturing Snapshot of Trends Up 1`] = `
<table
  className="MuiTable-root-18 Trends-table-9"
>
  <thead
    className="MuiTableHead-root-19 Trends-tableRow-10"
  >
    <tr
      className="MuiTableRow-root-20 MuiTableRow-head-23 Trends-tableRow-10"
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Customer
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Percentage Behind
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-head-26 Trends-tableCell-12"
        scope="col"
      >
        Trend
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root-37"
  >
    <tr
      className="MuiTableRow-root-20 Trends-tableRow-10"
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        ABC
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableCell-12"
      >
        1.0
        %
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root-25 MuiTableCell-body-27 Trends-tableActions-11"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-38 undefined undefined"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </td>
    </tr>
  </tbody>
</table>
`;
